app/assets/javascripts/discourse/app/services/login.js

```diff
@@ -16,4 +19,20 @@ export default class LoginService extends Service {
       setLoggingIn?.(false);
     }
   }
+
+  @action
+  async singleExternalLogin() {
+    await this.externalLogin(this.externalLoginMethods[0]);
+  }
+
+  get isOnlyOneExternalLoginMethod() {
+    return (
+      !this.siteSettings.enable_local_logins &&
+      this.externalLoginMethods.length === 1
+    );
+  }
+
+  get externalLoginMethods() {
+    return findAll();
+  }
```

Comment:
Would it make sense to use `singleExternalLogin()` from this service in `routes/application.js` as well?